app/models/user.rb

```diff
@@ -408,15 +409,15 @@ def generate_token
 
   class << self
     def search(query)
-      query = query.gsub('@',' ') # @ is a special char in full text search in MYSQL, and cannot be escaped; https://github.com/publiclab/plots2/issues/8344
-      query = query + '*' unless query.empty?
-      User.where('MATCH(bio, username) AGAINST(? IN BOOLEAN MODE)', "#{query}")
+      query = query.tr('@', ' ') # @ is a special char in full text search in MYSQL, and cannot be escaped; https://github.com/publiclab/plots2/issues/8344
+      query += '*' unless query.empty?
+      User.where('MATCH(bio, username) AGAINST(? IN BOOLEAN MODE)', query.to_s)
```

Comment:
A good practice that improves the readability of your code is breaking a line between the body of the function and their response.

```rb
def search(...)
   query = query.tr('@', ' ')
   query += '*' unless query.empty?
   
   User.where('MATCH(bio, username) AGAINST(? IN BOOLEAN MODE)', query)
end
```
When another developer reads, they will be aware that this is the implied return of the function.

Reply:
btw, I think you don't need the `.to_s` in the query call!

Reply:
Codeclimate reported to use `to_s` instead of string interpolation, however, I've now removed it.